File: oa-portal/problems/m27-warehouse-robot-keys/starters/main.cpp
#include <bits/stdc++.h>
using namespace std;

/* grid[i][j] is one of . # S X a-f A-F. Return -1 if X cannot be reached. */
int fewestMoves(const vector<string>& grid) {
    // write your code here

    return -1;
}

int main() {
    int r, c;
    if (scanf("%d %d", &r, &c) != 2) return 0;
    vector<string> grid(r);
    for (int i = 0; i < r; i++) {
        char buf[128];
        scanf("%s", buf);
        grid[i] = buf;
    }

    printf("%d\n", fewestMoves(grid));
    return 0;
}

File: oa-portal/problems/m27-warehouse-robot-keys/starters/main.java
import java.io.*;
import java.util.*;

public class Main {

    /* grid[i].charAt(j) is one of . # S X a-f A-F. Return -1 if X is unreachable. */
    static int fewestMoves(String[] grid) {
        // write your code here

        return -1;
    }

    public static void main(String[] args) throws IOException {
        FastReader in = new FastReader();
        int r = in.nextInt(), c = in.nextInt();
        String[] grid = new String[r];
        for (int i = 0; i < r; i++) grid[i] = in.next();

        System.out.println(fewestMoves(grid));
    }

    static class FastReader {
        private final DataInputStream in = new DataInputStream(new BufferedInputStream(System.in, 1 << 16));
        int nextInt() throws IOException { return (int) nextLong(); }
        long nextLong() throws IOException {
            int b = in.read();
            while (b != '-' && (b < '0' || b > '9')) b = in.read();
            boolean neg = b == '-';
            if (neg) b = in.read();
            long v = 0;
            while (b >= '0' && b <= '9') { v = v * 10 + (b - '0'); b = in.read(); }
            return neg ? -v : v;
        }
        String next() throws IOException {
            int b = in.read();
            while (b <= ' ') b = in.read();
            StringBuilder sb = new StringBuilder();
            while (b > ' ') { sb.append((char) b); b = in.read(); }
            return sb.toString();
        }
    }
}
